### PR TITLE
Update to new sklearn API

### PR DESCRIPTION
**Pull request by icfly2**
Change as per FutureWarning in sklearn, to allow eli5 to be used with sklearn 0.24 and newer. [docs](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.check_scoring.html)

Currently issues FutureWarning
```
  File "/app/model/model.py", line 3, in <module>
    import eli5
  File "/usr/local/lib/python3.7/site-packages/eli5/__init__.py", line 13, in <module>
    from .sklearn import explain_weights_sklearn, explain_prediction_sklearn
  File "/usr/local/lib/python3.7/site-packages/eli5/sklearn/__init__.py", line 3, in <module>
    from .explain_weights import (
  File "/usr/local/lib/python3.7/site-packages/eli5/sklearn/explain_weights.py", line 78, in <module>
    from .permutation_importance import PermutationImportance
  File "/usr/local/lib/python3.7/site-packages/eli5/sklearn/permutation_importance.py", line 15, in <module>
    from sklearn.metrics.scorer import check_scoring  # type: ignore
  File "/usr/local/lib/python3.7/site-packages/sklearn/metrics/scorer.py", line 12, in <module>
    _raise_dep_warning_if_not_pytest(deprecated_path, correct_import_path)
  File "/usr/local/lib/python3.7/site-packages/sklearn/utils/deprecation.py", line 143, in _raise_dep_warning_if_not_pytest
    warnings.warn(message, FutureWarning)
FutureWarning: The sklearn.metrics.scorer module is  deprecated in version 0.22 and will be removed in version 0.24. The corresponding classes / functions should instead be imported from sklearn.metrics. Anything that cannot be imported from sklearn.metrics is now part of the private API.
```

### CODE REVIEW

**Comment by jimbudarz** [comment]
This update will break compatibility with scikit-learn < 0.20, so it would be necessary to update the requirements and setup.py to require at least this version (eli5 currently allows >=0.18).

**Comment by icfly2** [comment]
Is the upgrade of the sklearn requirement acceptable, given that 0.18 is now 4 years old and 0.24 has a RC out?

**Comment by lopuhin** [comment]
> Is the upgrade of the sklearn requirement acceptable, given that 0.18 is now 4 years old and 0.24 has a RC out?

I'd say that yes, it is.

Thanks for the PR 👍

**Comment by veonua** [comment]
/eli5/sklearn/text.py in <module>
       
----> 4 from sklearn.feature_extraction.text import VectorizerMixin  # type: ignore

should be renamed to _VectorizerMixin

**Comment by veonua** [comment]
~/.local/lib/python3.8/site-packages/eli5/sklearn/transform.py 
----> 6 from sklearn.feature_selection.base import SelectorMixin  # type: ignore

from sklearn.feature_selection import SelectorMixin

**Comment by icfly2** [comment]
@veonua I'm a bit confused what you are suggesting, master reads:
```
try:
    from sklearn.feature_extraction.text import _VectorizerMixin as VectorizerMixin
except ImportError:  # Changed in scikit-learn 0.22
    from sklearn.feature_extraction.text import VectorizerMixin
```
your other comment is also already addressed in this PR

**Comment by veonua** [comment]
can you please publish these changes to pip?

**Comment by lopuhin** [comment]
Thanks you! This were merged in eli5-org/eli5#2 and released to PyPI with v0.11
